---
title: "WheatMicrobiomeRepro_analayis"
author: "Justice Ruwona"
date: "2023-04-10"
output: 
  html_document:
      keep_md: yes
---
# 1.0 Load in Libraries
```{r}
#load libraries
library(phyloseq)
library(tidyverse)
library(Biostrings)
library(ggplot2)
library(vegan)
library(dplyr)
library(lme4)
library(emmeans)
library(multcomp)
library(ggpubr)
```
```{r}
##### Set global options #####

# no scientific notation
options(scipen=10000) 

# color blind pallet  
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

```
# 2.0 READ IN DATA STEPS
```{r}
###### Read in data ####

#Loading the mapping file metadata
samp_dat <- read.csv("Data_files/mapping_its.csv")
samp_dat <- samp_dat %>%
  filter(Crop == "Yr_1") # filter only wheat dataset, exclude other crops in yr2, yr3 of rotation
rownames(samp_dat) <- samp_dat$Sample_Name # format row names must match OTU table headers
SAMP.fungi <- phyloseq::sample_data(samp_dat)

# OTU table - 
otu <- read.csv("Data_files/otu_table_its.csv")
treatment <- factor(c("T1", "T2", "T3", "T4"))
rownames(otu) <- otu$OTU_ID # format row names must match OTU table headers
otu <- otu[,-1] # shifting to 1, numeric matrix
OTU.fungi <- phyloseq::otu_table(otu, taxa_are_rows = TRUE)

# Taxonomy table and choose Fungi only dataset
taxonomy.fungi <- read.csv("Data_files/taxonomy_table_its.csv")
rownames(taxonomy.fungi) <- taxonomy.fungi$OTU_ID
taxonomy.fungi2 <- taxonomy.fungi %>%
  subset(Kingdom == "Fungi")

TAX.fungi <- phyloseq::tax_table(as.matrix(taxonomy.fungi2))

# Fasta
#FASTA.fungi <- readDNAStringSet("ITS_all_crops.fasta", format="fasta", seek.first.rec=TRUE, use.names=TRUE)

```
### Create Phyloseq object 
```{r}
###### Create Initial Phyloseq object ######
fungi.unedited <- phyloseq::phyloseq(OTU.fungi, TAX.fungi, SAMP.fungi)


```

### Prune sample reads and run reads statistics
```{r}
## Remove samples with less than 5000 reads
fungi.edited <- fungi.unedited %>% 
  prune_samples(sample_sums(.) > 5000, .) %>% # remove samples below 5,000 reads
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) # remove taxa with less than 1 reads
sample_sums(fungi.edited) %>%
  sort()
# New number of total reads
sum(sample_sums(fungi.edited)) # should read 4,272,434
# Mean and median read depth 
mean(sample_sums(fungi.edited)) #  should read 20,443.21
median(sample_sums(fungi.edited)) #  should read 16,512
# Histogram including median read depth
read.depths <- data.frame(sample_sums(fungi.edited))
colnames(read.depths) <- "read.depth"
read.depth.plot <- ggplot(read.depths, aes(read.depth)) +
  geom_histogram(fill = cbbPalette[[5]], color = "black") + 
  geom_vline(xintercept = median(sample_sums(fungi.unedited)), linetype = "dashed") + 
  theme_classic() + 
  ggtitle("Reads Depth Histogram") +
  xlab("Read Depth") + ylab("Count")
read.depth.plot
```

# 3.0 DATA ANALYSIS
```{r}
# Global fungal community  dataset combining growth stage, plant organ, and crop management strategy 
subset_phylo <- subset_samples(fungi.edited)

```

## 3.1 Alpha diversity for global fungal endophytic communities
```{r}
# Rename fungi.edited data
physeq_sub <- fungi.unedited 
      
# Calculate alpha diversity metrics (Shannon, Simpson, and Observed OTUs) and print table
alpha_div <- estimate_richness(physeq_sub, measures=c("Shannon", "Simpson", "Observed"))
# Add treatment, collection time interval and tissue information to alpha diversity table
alpha_div <- cbind(Treatment = as.factor(sample_data(physeq_sub)$Treatment), 
                   Collection = as.factor(sample_data(physeq_sub)$Collection),
                   Tissue = as.factor(sample_data(physeq_sub)$Tissue),
                   alpha_div)
# Calculate summary statistics for each alpha diversity metric 
alpha_div_summary <- alpha_div %>%
  group_by(Treatment, Collection, Tissue) %>%
  summarize(mean_shannon = mean(Shannon), 
            sd_shannon = sd(Shannon), 
            mean_simpson = mean(Simpson), 
            sd_simpson = sd(Simpson), 
            mean_observed = mean(Observed), 
            sd_observed = sd(Observed))
alpha_div_summary # view summary of alpha diversity
```

### Using whole plant tissue (Global community) Shannon index to compare if management communities are different
```{r}
#### ANOVA based on Shannon index
lm.alphadiv <- (lm(mean_shannon ~ Treatment*Collection,data = alpha_div_summary))
anova(lm.alphadiv)

### Test ls means using Tukey
smeans <- emmeans(lm.alphadiv, ~ Treatment | Collection) # estimate lsmeans of Treatment within Collection
Results_lsmeans <- cld(smeans, alpha = 0.05, reversed = TRUE, details = TRUE) # contrast with Tukey adjustment
Results_lsmeans # expect to see Collection 3 T2 - T4   p = 0.0233 is Significant

#filter significant comparisons
lsd_df <- as.data.frame(Results_lsmeans$comparisons) # Table form  
signif_lsmean <- lsd_df[lsd_df$p.value< 0.05,] # only  Collection 3 T2 - T4   p = 0.0233 is Significant
signif_lsmean
```
### Boxplots to summarize Alpha diversity (mean _shannon) based on global fungal community

```{r}
# Create the box plot Treatments
ggplot(alpha_div_summary, aes(x=Treatment, y=mean_shannon, fill=Collection)) + 
  geom_boxplot() +
  scale_fill_manual(values=c( "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7"),
                    labels=c("Vegetative", "Flowering", "Seed development")) +
  scale_x_discrete(labels=c("Conventional", "No till", "Low Input", "Organic")) +
  xlab("Management") + ylab("Shannon diversity index") +
  ggtitle("Alpha Diversity Boxplot")+
  theme_bw() +
  theme(panel.background = element_rect(fill = "white"),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank()) 
  # plot1 + geom_text(data=as.data.frame(signif_lsmean),
            #aes(x = treatment, y = mean_shannon , label=letters[1:nrow(signif_lsmean)]))
  #geom_text(data=as.data.frame(signif_lsmean), aes(x = Contrast, y = emmean, label=letters))
  
 

```

## 3.2 Beta diversity for global community
```{r}
# Beta diversity 
fungi.dist.bray = phyloseq::distance(fungi.edited, "bray") # create bray-curtis distance matrix
fungi.ord <- ordinate(fungi.edited, "PCoA", "bray")
global.nmds <- plot_ordination(fungi.edited, ordination = fungi.ord, type = "samples") 
global.nmds.data <- global.nmds$data

# Permanova of communities considering all factors and their interactions
permanova <- adonis2(fungi.dist.bray~Collection*as.factor(Tissue)*as.factor(Treatment), as(sample_data(fungi.edited), "data.frame"), permutations = 9999) 

#beta diversity plot: Principal Component chart
ggplot() + 
  geom_point(data = global.nmds.data, aes(x = Axis.1, y = Axis.2, shape = as.factor(Tissue), fill = as.factor(Treatment)), alpha = 0.8, size = 2) +
  theme_bw() +
  ylab("PCoA2") + 
  xlab("PCoA1") +
  scale_fill_manual(values=cbbPalette) +
  stat_ellipse(data = global.nmds.data, aes(x = Axis.1, y = Axis.2, group = Treatment), type = "norm", linetype = 2) +
  scale_shape_manual(values=c(21, 22, 23, 24, 25)) +
  guides(fill=guide_legend(override.aes=list(shape=21))) 
```


